### ISSUE TITLE
[RFC] Extract default code component properties

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#991]
### What's the problem? 🤔

When copying a demo from the MUI docs, none of the properties are bindable. This makes the component less usable out of the box as it can't be interacted with on the page. Currently the user has to add this functionality themselves after importing by wrapping it with `createComponent`.

### Proposed solution 🟢

When a user creates a component that hasn't been wrapped with `createComponent` we will generate a default `argTypes` definition. We will try to extract it from:

- `propTypes` if they exist on the component
- the typescript types

As soon as the user ads `createComponent` we will ignore this default.

The extraction needs to happen at build time, when the component is defined, not at runtime. Reasons are:

- In future optimizations, the runtime won't have type information attached
- It will be too expensive, the bundle will need to contain a full parser to extract the information from the source

We can likely try using [`extract-react-types`](https://github.com/atlassian/extract-react-types/tree/master/packages/extract-react-types) fro this. It looks like it can be made to [work in the browser](https://extract-react-types.com/repl).

**Comment by prakhargupta1** [comment]
Not sure if it relates to it, but I encountered a use case today. I imported MUI pagination component and wanted to use the current page number prop from this custom component into my application to render some assets, but couldn't do it.

**Comment by Janpot** [comment]
> I imported MUI pagination component and wanted to use the current page number prop from this custom component into my application to render some assets, but couldn't do it.

Would the following work for you?

```tsx
import * as React from "react";
import Pagination from "@mui/material/Pagination";
import Stack from "@mui/material/Stack";
import { createComponent } from "@mui/toolpad-core";

interface BasicPaginationProps {
  count: number;
  page: number;
  onChange: (value: number) => void;
}

function BasicPagination({ count, page, onChange }: BasicPaginationProps) {
  const handleChange = React.useCallback(
    (event, newPage) => onChange(newPage),
    [onChange]
  );
  return (
    <Stack spacing={2}>
      <Pagination count={count} page={page} onChange={handleChange} />
    </Stack>
  );
}

export default createComponent(BasicPagination, {
  argTypes: {
    count: {
      typeDef: { type: "number" },
      defaultValue: 10,
    },
    page: {
      typeDef: {
        type: "number",
      },
      onChangeProp: "onChange",
    },
  },
});
```

**Comment by prakhargupta1** [comment]
Yes, it does. Thanks. App [link](https://master--toolpad.mui.com/_toolpad/app/cl7ucighk0151axn52oz4s88h/pages/hs1a55l)
